app/Http/Controllers/ProductController.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Services\ProcessProductFileServiceInterface;
+
+class ProductController extends Controller
+{
+    private ProcessProductFileServiceInterface $fileService;
+
+    /**
+     * @param ProcessProductFileServiceInterface $fileService
+     */
+    public function __construct(ProcessProductFileServiceInterface $fileService)
+    {
+        $this->fileService = $fileService;
+    }
```

Comment:
https://php.watch/versions/8.0/constructor-property-promotion
Could be refactored as 
```suggestion
    public function __construct(private ProcessProductFileServiceInterface $fileService)
    {
    }
```